class-two-factor-core.php

```diff
@@ -925,6 +925,11 @@ public static function manage_users_custom_column( $output, $column_name, $user_
 	 * @param WP_User $user WP_User object of the logged-in user.
 	 */
 	public static function user_two_factor_options( $user ) {
+		
+		/* workaround for two factor options crashing bbpress' profile page */
+		if ( function_exists('bbp_is_user_home_edit') && bbp_is_user_home_edit() ) return;
```

Comment:
This is currently relying on a specific integration and historically this plugin doesn't have any conditional logic to account for specific integrations. Could we somehow convert this into something that can be placed in the `class-two-factor-compat.php` file which currently deals with Jetpack specific things?

https://github.com/WordPress/two-factor/blob/master/class-two-factor-compat.php

Which of the functionality in this method is actually not compatible with the front-end output?

Reply:
You might want to check out my comment where I outline frontend integration with this problem (and a few more) here: https://github.com/WordPress/two-factor/issues/335#issuecomment-719087800

For this specific issue, it would probably be better to redeclare the admin functions on the frontend when needed.

Reply:
> 
> 
> This is currently relying on a specific integration and historically this plugin doesn't have any conditional logic to account for specific integrations. Could we somehow convert this into something that can be placed in the `class-two-factor-compat.php` file which currently deals with Jetpack specific things?
> 
> https://github.com/WordPress/two-factor/blob/master/class-two-factor-compat.php
> 
> Which of the functionality in this method is actually not compatible with the front-end output?

I updated the PR to include an example error screenshot.
Until the required admin functions are exposed on the frontend (should they?), it might be possible to rewrite the workaround in the compat file to conditionally remove the two action hooks instead. I'll look into this.